doc/tutorials/electrodes/electrodes_part1.ipynb

```diff
@@ -0,0 +1,500 @@
+    "All physical parameters are defined in reduced units of length ($\\sigma=1$;\n",
+    "Particle size), mass ($m=1$; Particle mass), time ($t=0.01 \\tau$) and\n",
+    "elementary charge ($e=1$).\n",
+    "\n",
+    "Another important length scale is the Bjerrum Length, which is the length at\n",
+    "which the electrostatic energy between two elementary charges is comparable to\n",
+    "the thermal energy $k_\\mathrm{B}T$ and thus defines the energy scale in our\n",
+    "system.\n",
+    "It is defined as\n",
+    "$$\\ell_\\mathrm{B}=\\frac{1}{4\\pi\\epsilon_0\\epsilon_r k_\\mathrm{B}T}.$$ \n",
+    "In our case, if we choose the ion size ($\\sigma$) in simulations to represent a\n",
+    "typical value for mono-atomar salt, 0.3 nm in real units, then the\n",
+    "Bjerrum length of water at room temperature, $\\ell_\\mathrm{B}=0.71 \\,\\mathrm{nm}$ is\n",
+    "$\\ell_\\mathrm{B}\\sim 2$ in simulations units."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#***************************************************\n",
+    "#            System Setup\n",
+    "#***************************************************\n",
+    "\n",
+    "# Box dimensions\n",
+    "# To construct a narrow slit Lz << (Lx , Ly)\n",
+    "box_l_x = 100.\n",
+    "box_l_y = 100.\n",
+    "box_l_z = 5.\n",
+    "\n",
+    "# ICC* with ELC: 2D electrostatics\n",
+    "ELC_GAP = 6*box_l_z\n",
+    "\n",
+    "system = espressomd.System(box_l=[box_l_x, box_l_y, box_l_z+ELC_GAP])\n",
+    "\n",
+    "# System Time\n",
+    "system.time_step = 0.01\n",
+    "system.cell_system.skin = 0.4\n",
+    "\n",
+    "# Elementary charge \n",
+    "q = np.array([1.0])  \n",
+    "\n",
+    "# Interaction Parameters for P3M with ELC\n",
+    "\n",
+    "BJERRUM_LENGTH = 2.0        # Electrostatic prefactor passed to P3M ; prefactor=lB KBT/e2                \n",
+    "ACCURACY = 1e-7             # P3M force accuracy      \n",
+    "CHECK_ACCURACY = 1e-7       # maximim pairwise error in ELC\n",
+    "\n",
+    "#Lennard-Jones  Parameters\n",
+    "\n",
+    "LJ_SIGMA = 1.0\n",
+    "LJ_EPSILON = 1.0                \n",
+    "\n",
+    "#Particle parameters\n",
+    "\n",
+    "types = {\"Cation\": 0, \"Anion\": 1  ,\"Electrodes\": 2}\n",
+    "charges = {\"Cation\": q[0], \"Anion\": -q[0]  }\n",
+    "\n",
+    "p1=system.part.add(pos=[box_l_x/4.0, box_l_y/2.0, box_l_z/2.0], q=charges[\"Cation\"])\n",
+    "print(f\"Cation placed at position: {p1.pos}\")\n",
+    "p2=system.part.add(pos=[3.0*box_l_x/4.0, box_l_y/2.0, box_l_z/2.0], q=charges[\"Anion\"])\n",
+    "print(f\"Anion placed at position: {p2.pos}\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "solution2": "hidden",
+    "solution2_first": true
+   },
+   "source": [
+    "### Task:\n",
+    "Set up electrostatics interactions between the ion pairs using P3M and ICC $\\star$.  \n",
```

Comment:
this task is very long. It would be better to split it in smaller parts and take parts out of the task. especially because variables defined in the solution (but not mentioned in the task) are used further down